test/src/monitoring_frame_serialization.cpp

```diff
@@ -25,30 +25,37 @@ DynamicSizeRawData serialize(MonitoringFrameMsg& frame)
 {
   std::ostringstream os;
 
-  raw_processing::write(os, frame.device_status_fixed_);
-  raw_processing::write(os, frame.op_code_fixed_);
-  raw_processing::write(os, frame.working_mode_fixed_);
-  raw_processing::write(os, frame.transaction_type_fixed_);
-  raw_processing::write(os, frame.scanner_id_fixed_);
-  raw_processing::write(os, frame.from_theta_fixed_);
-  raw_processing::write(os, frame.resolution_fixed_);
-
-  FieldHeader::Id scan_counter_header_id = AdditionalFieldIds::SCAN_COUNTER;
-  FieldHeader::Length scan_counter_header_length = sizeof(frame.scan_counter_) + 1;
+  raw_processing::write(os, DEFAULT_DEVICE_STATUS);
+  raw_processing::write(os, OP_CODE_MONITORING_FRAME);
+  raw_processing::write(os, ONLINE_WORKING_MODE);
+  raw_processing::write(os, GUI_MONITORING_TRANSACTION);
+  raw_processing::write(os, frame.scanner_id_);
+  raw_processing::write(os, frame.from_theta_.value());
+  raw_processing::write(os, frame.resolution_.value());
+
+  MonitoringFrameAdditionalFieldHeader scan_counter_header(MonitoringFrameAdditionalFieldIds::SCAN_COUNTER,
+                                                           sizeof(frame.scan_counter_));
+  writeFieldHeader(os, scan_counter_header);
   uint32_t scan_counter_header_payload = frame.scan_counter_;
-  raw_processing::write(os, scan_counter_header_id);
-  raw_processing::write(os, scan_counter_header_length);
   raw_processing::write(os, scan_counter_header_payload);
 
-  FieldHeader::Id measures_header_id = AdditionalFieldIds::MEASURES;
-  FieldHeader::Length measures_header_length = frame.measures_.size() * NUMBER_OF_BYTES_SINGLE_MEASURE + 1;
-  raw_processing::write(os, measures_header_id);
-  raw_processing::write(os, measures_header_length);
+  if (frame.diagnostic_data_enabled_)
+  {
+    MonitoringFrameAdditionalFieldHeader diagnostic_data_field_header(MonitoringFrameAdditionalFieldIds::DIAGNOSTICS,
+                                                                      DIAGNOSTIC_DATA_LENGTH_IN_BYTES);
+    writeFieldHeader(os, diagnostic_data_field_header);
+    std::array<uint8_t, DIAGNOSTIC_DATA_LENGTH_IN_BYTES> diagnostic_data_field_payload =
+        serializeDiagnosticMessages(frame.diagnostic_messages_);
+    raw_processing::write(os, diagnostic_data_field_payload);
+  }
```

Comment:
What if this is true but there are no diagnostic messages?

Reply:
The function should write out the raw data with the diagnostic data block, although there will be only zeros in the 40 bytes.